botorch/acquisition/proximal.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+A wrapper around AquisitionFunctions to fix certain features for optimization.
+This is useful e.g. for performing contextual optimization.
+"""
+
+from __future__ import annotations
+
+import torch
+from botorch.acquisition.analytic import AnalyticAcquisitionFunction
+from botorch.utils import t_batch_mode_transform
+from torch import Tensor
+from torch.nn import Module
+
+
+class ProximalAcquisitionFunction(AnalyticAcquisitionFunction):
+    """A wrapper around AquisitionFunctions to add proximal weighting of the acquisition function. Acquisition
+    function is weighted via a squared exponential centered at the last training point, with varying lengthscales
+    corresponding to 'proximal_weights'. Can only be used with single batch analytical acquisition functions.
```

Comment:
Since this can only be used with single batch analytic acquisition functions, should we add checks for these in the constructor?

Just curious, why is this limited to analytic acquisition functions? Does anything break if I were to use this with an MC acquisition function (along with SAA to fix base samples - botorch default)?

Reply:
Turns out it is not limited to single batch analytic acquisition functions, just single batch models (see additions to test_proximal